kyaml/filesys/filesystem_test.go

```diff
@@ -42,3 +52,87 @@ func testNotExistErr(t *testing.T, fs FileSystem) {
 	err = fs.Walk(path, func(_ string, _ os.FileInfo, err error) error { return err })
 	assert.Truef(t, errors.Is(err, os.ErrNotExist), "Walk should return ErrNotExist, got %v", err)
 }
+
+// setupFileSys returns file system, default directory to test in, and working directory
+func setupFileSys(t *testing.T, name string) (FileSystem, string, string) {
+	t.Helper()
+
+	switch name {
+	case disk:
+		fSys, testDir := makeTestDir(t)
+		return fSys, testDir, cleanWd(t)
+	case memoryAbs:
+		return filesysBuilders[name](), Separator, Separator
+	default:
```

Comment:
Nit: I see the default case is intended to be the `memoryEmpty` case. For these types of switch statements, it's better to have a case for every possible value that you support, and have `default` be an error case.

i.e.

```
	switch name {
	case disk:
		fSys, testDir := makeTestDir(t)
		return fSys, testDir, cleanWd(t)
	case memoryAbs:
		return filesysBuilders[name](), Separator, Separator
	case makeEmpty:
		return filesysBuilders[name](), "", "
   default:
        // return some error about how `name` is an invalid argument type
```

Reply:
Agree that `memoryEmpty` should have its own `case`. However, I would ideally like this test helper function to take care of any errors on its own instead of cluttering the actual test function.

I changed the code to look like this: https://github.com/kubernetes-sigs/kustomize/pull/4677/files#diff-b0f1942e0058e910ccc91dc07e75960a2cef265b14715ed4e236e69c6cafbf22R68. Will resolve to throw an error if this is bad practice.